### PR TITLE
Add possibility to disable slider after initialization.

### LINKED ISSUE #8497: Problem disabling slider with jquery
I have an issue with the foundation slider described here:
http://foundation.zurb.com/sites/docs/slider.html
When I add the "disabled" class to the slider directly in HTML it gets disabled as expected. No dragging or clickSelecting of the handle is possible.

But when I do this: `$("#my_slider").addClass("disabled");` the slider gets disabled but it's still possible to drag and clickSelect the slider handle.

Anyone?

Jimmy

### PR DESCRIPTION
**Pull request by Owlbertz**
Slider will check for disabled state on every event now rather than only at its initialization.

Fixes #8497.

### CODE REVIEW

**Comment by ncoden** [comment]
Thanks. Should we check the `[disabled=false]` case ?

**Comment by Owlbertz** [comment]
Yeah, you're absolutely right. It only makes sense if you can also enable a disabled slider... Did another commit for this.

**Comment by ncoden** [comment]
I mean : should we consider the slider as enabled when it has a `disabled="false"` or `disabled="enabled"` attribute ?

**Comment by ncoden** [comment]
I don't know if `this.options.disabled` care about this cases, or if it is only checking that a `disabled` attribute has been set, regardless of its value

**Comment by Owlbertz** [comment]
I guess you mean `data-disabled`, since `disabled` is only for `input`s. The options are only initialized once based on this attribute, so for now only checking the class is possible.

File: js/foundation.slider.js [hunk 1]
```diff
@@ -106,7 +106,11 @@ class Slider {
    * @fires Slider#changed
    */
   _setHandlePos($hndl, location, noInvert, cb) {
-  //might need to alter that slightly for bars that will have odd number selections.
+    // don't move if the slider has been disabled since its initialization
+    if (this.$element.hasClass(this.options.disabledClass)) {
```
Comment (ncoden) [inline js/foundation.slider.js]:
```js
var disabledAttr = this.$element.data('disabled');
var disabledClass = this.$element.hasClass(this.options.disabledClass);

if ((disabledClass || disabledAttr) && !(disabledAttr && disabledAttr == "false")) {
```

So :
- `disabled` class or attribute (not `false`) -> **Disabled**
- `disabled` option in initialization -> **Disabled**
- `disabled` option in initialization + `disabled` class or attribute (not `false`) -> **Disabled**
- `disabled` option in initialization + `disabled` class + `disabled` attribute equal to `false` -> **Enabled**

Comment (kball) [inline js/foundation.slider.js]: I'd prefer there be a single source of truth, and it looks like we're currently using the class as the way to do that.  @ncoden did you see someplace in particular that is setting data-disabled?  I'm not seeing that mechanism being used.

**Comment by kball** [comment]
@Owlbertz this looks good.  Can you set up a unit test for this case in test/javascript/components/slider.js? You can use the tests in test/javascript/components/toggler.js as an example of how these can be structures. See: https://github.com/zurb/foundation-sites/blob/develop/test/javascript/components/toggler.js

Thanks!

File: js/foundation.slider.js (hunk 1 above)
Comment (ncoden) [inline js/foundation.slider.js]: @kball `this.options`*`.enabled`* is initialized with `$element.data`*`.enabled`*

**Comment by kball** [comment]
Gonna move forward with this

**Pull request merged**